crypto/evp/evp_lib.c

```diff
@@ -404,7 +404,7 @@ int EVP_MD_meth_get_app_datasize(const EVP_MD *md)
 }
 unsigned long EVP_MD_meth_get_flags(const EVP_MD *md)
 {
-    return md->block_size;
+    return md->flags;
```

Comment:
Yeah, I was thinking this might need to be submitted regardless of the state of the rest of the PR.